ksql-engine/src/main/java/io/confluent/ksql/materialization/ks/KsMaterializedWindowTable.java

```diff
@@ -47,23 +47,31 @@
   @Override
   public List<WindowedRow> get(
       final Struct key,
-      final Range<Instant> windowStart
+      final Range<Instant> windowStartBounds
   ) {
     try {
       final ReadOnlyWindowStore<Struct, GenericRow> store = stateStore
           .store(QueryableStoreTypes.windowStore());
 
-      final Instant lower = windowStart.lowerEndpoint();
-      final Instant upper = windowStart.upperEndpoint();
+      final Instant lower = windowStartBounds.hasLowerBound()
+          ? windowStartBounds.lowerEndpoint()
+          : Instant.ofEpochMilli(Long.MIN_VALUE);
+
+      final Instant upper = windowStartBounds.hasUpperBound()
+          ? windowStartBounds.upperEndpoint()
+          : Instant.ofEpochMilli(Long.MAX_VALUE);
 
       try (WindowStoreIterator<GenericRow> it = store.fetch(key, lower, upper)) {
 
         final Builder<WindowedRow> builder = ImmutableList.builder();
 
         while (it.hasNext()) {
           final KeyValue<Long, GenericRow> next = it.next();
-          final Window window = Window.of(Instant.ofEpochMilli(next.key), Optional.empty());
-          builder.add(WindowedRow.of(stateStore.schema(), key, window, next.value));
+          final Instant windowStart = Instant.ofEpochMilli(next.key);
+          if (windowStartBounds.contains(windowStart)) {
```

Comment:
is this to check the case where the bound is open and the window start is on the boundary?

Reply:
Yep.